src/run.ts

```diff
@@ -283,3 +156,78 @@ export function onDidStartDebugSession(e: vscode.DebugSession) {
     console.warn("Failed to open browser", err);
   });
 }
+
+async function createTerminalAndCloseOthersWithSameName(
+  options: vscode.TerminalOptions
+): Promise<vscode.Terminal> {
+  if (!options.name) {
+    throw new Error("Terminal name is required");
+  }
+
+  // Grab a list of all terminals with the same name, before we create the new
+  // one. We'll close them. (It'd be surprising if there was more than one,
+  // given that we always close the previous ones when starting a new one.)
+  const oldTerminals = vscode.window.terminals.filter(
+    (term) => term.name === options.name
+  );
+
+  const newTerm = vscode.window.createTerminal(options);
+  newTerm.show(true);
+
+  // Wait until new terminal is showing before disposing the old ones.
+  // Otherwise we get a flicker of some other terminal in the in-between time.
+
+  const closingTerminals = oldTerminals.map((x) => {
+    const p = new Promise<void>((resolve) => {
+      // Resolve when the terminal is closed
+      const subscription = vscode.window.onDidCloseTerminal(function sub(term) {
+        if (term === x) {
+          subscription.dispose();
```

Comment:
Add comment that the ports aren't always (or even usually) closed at this point, for some reason